internal/updater/updater.go

```diff
@@ -259,14 +276,14 @@ func (u *UpdateInstaller) InstallWithProgress(installTargetPath string, progress
 	return proc, nil
 }
 
-func (u *UpdateInstaller) analyticsEvent(action, label, version, msg string) {
+func (u *UpdateInstaller) analyticsEvent(action, label, msg string) {
 	dims := &dimensions.Values{
-		TargetVersion: ptr.To(version),
+		TargetVersion: ptr.To(u.AvailableUpdate.Version),
```

Comment:
This probably needs a nil check. eg. the `ActUpdateCheck` event won't have an available update yet.

Reply:
`ActUpdateCheck` happens in the `checker.go` file as individual events: https://github.com/ActiveState/cli/blob/mitchell/dx-2128/internal/updater/checker.go#L122 and https://github.com/ActiveState/cli/blob/mitchell/dx-2128/internal/updater/checker.go#L139. This code happens in an `UpdateInstaller` method: https://github.com/ActiveState/cli/blob/mitchell/dx-2128/internal/updater/updater.go#L279, whose `AvailableUpdate` member is never `nil` from what I can see. Furthermore, this change is just a simplification, as all calls to `analyticsEvent()` were already dereferencing `u.AvailableUpdate`, e.g. https://github.com/ActiveState/cli/pull/2781/files/5f4ebc6c75ce6f701be009e7a111f26969cba933#diff-07f86aa363d0be00f9de8b4134d96188f3923b9557bc3428c65c8541027ea68dL135

Reply:
That's fair, but all the same for future compatibility I'd rather we nil check this just to be safe. The function this is implemented in could in the future be used for events that may not have the AvailableUpdate set.